tests/test_advanced_options.py

```
@@ -288,6 +288,23 @@ async def _side(opt: str, *args: object, **kwargs: object) -> object:
     assert "zone_name" in called
 
 
+@pytest.mark.asyncio
+async def test_build_next_option_only_traverses_comma_prefixed_suffix(sensor: MockSensor) -> None:
+    """Do not process malformed non-comma suffix text after a bracket option."""
+    sensor.attrs = {"devicetracker_zone_name": "Home", "place_type": "Restaurant"}
+    parser = AdvancedOptionsParser(sensor, "zone_name[place_type]place_type")
+    calls: list[str] = []
+
+    async def _side(opt: str, *args: object, **kwargs: object) -> object:
```

Ruff (0.15.13)

[warning] Unused function argument: `args`

(ARG001)